spec/migrations/20240521100629_add_vault_credential_id_to_options_in_service_templates_spec.rb

```diff
@@ -0,0 +1,43 @@
+require_migration
+
+describe AddVaultCredentialIdToOptionsInServiceTemplates do
+  let(:service_template_stub) { migration_stub(:ServiceTemplate) }
+
+  migration_context :up do
+    it "adds vault_credential_id to provision and retirement options" do
+      service_template = service_template_stub.create!(:options => service_template_up_options)
+      migrate
+      expect(service_template.reload.options[:config_info][:provision][:vault_credential_id]).to eq('provision_credentials')
+      expect(service_template.reload.options[:config_info][:retirement][:vault_credential_id]).to eq('retirement_credentials')
+    end
```

Comment:
Similarly here, we should verify that the vault_credentials are gone.

```suggestion
    it "converts vault_credentials to vault_credential_id in provision and retirement options" do
      service_template = service_template_stub.create!(:options => service_template_up_options)
      migrate
      expect(service_template.reload.options[:config_info][:provision]).to_not have_key(:vault_credentials)
      expect(service_template.reload.options[:config_info][:provision][:vault_credential_id]).to eq('provision_credentials')
      expect(service_template.reload.options[:config_info][:retirement].to_not have_key(:vault_credentials)
      expect(service_template.reload.options[:config_info][:retirement][:vault_credential_id]).to eq('retirement_credentials')
    end
```